xdsl/dialects/affine.py

```diff
@@ -350,7 +357,7 @@ class Yield(IRDLOperation):
     name = "affine.yield"
     arguments: VarOperand = var_operand_def(AnyAttr())
 
-    traits = frozenset([IsTerminator()])
+    traits = frozenset([IsTerminator(), Pure()])
```

Comment:
I see on the MLIR page that it's ConditionallySpeculatable and NoMemoryEffect, is that the same as Pure, effectively?

Reply:
Ah, no, Pure is AlwaysSpeculatable and NoMemoryEffect
Thanks, will have a quick look at the speculatability implementation there 🤔

Reply:
Eeeerm
It is not marked as Pure in the docs: https://mlir.llvm.org/docs/Dialects/Affine/#affineyield-affineaffineyieldop
But it is Pure as per the code: https://github.com/llvm/llvm-project/blob/9cf052df90418697c05810f69c588064f7b3ce71/mlir/include/mlir/Dialect/Affine/IR/AffineOps.td#L901
:melting_face: 

I'd just consider it pure myself. does that sound right to you?